ui/page/wallets/wallet_page.go

```diff
@@ -1079,6 +1085,61 @@ func (pg *WalletPage) deleteBadWallet(badWalletID int) {
 		}).Show()
 }
 
+func (pg *WalletPage) listenForTxNotifications() {
+	go func() {
+		for {
+			var notification interface{}
+
+			select {
+			case notification = <-pg.Receiver.NotificationsUpdate:
+			case <-pg.ctx.Done():
+				return
+			}
+
+			switch n := notification.(type) {
+			case wallet.NewTransaction:
+				// refresh wallets when new transaction is received
+				pg.UpdateAccountBalance()
+				pg.RefreshWindow()
+			case wallet.SyncStatusUpdate:
+				// refresh wallet account and balance on every new block
+				// only if sync is completed.
+				if n.Stage == wallet.BlockAttached && pg.WL.MultiWallet.IsSynced() {
+					pg.UpdateAccountBalance()
+					pg.RefreshWindow()
+				}
+			}
+		}
+	}()
+}
+
+func (pg *WalletPage) UpdateAccountBalance() {
```

Comment:
Private method if not called outside this package. Doesn't look like it is called outside.